GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace GenericsHomework
+{
+    public class Node<T> where T : notnull
+    {
+        private T Value { get; set; }
+        public Node<T> Next { get;  private set; }
+
+        public Node(T value)
+        {
+            Value = value;
+            Next = this;
+        }
+        public override string? ToString()
+        {
+            return Value.ToString();
+        }
+        public void Append(T value)
+        {
+            if (Exists(value))
+                throw new ArgumentException("Value already exists in list.");
```

Comment:
```suggestion
                throw new ArgumentException($"{nameof(value)} already exists in list.");
```
Consider using `nameof` here